src/databricks/labs/remorph/snow/sql_transpiler.py

```diff
@@ -48,6 +48,27 @@ def parse_sql_content(self, sql, file_name):
                 for select in expr.find_all(exp.Select, exp.Join, exp.With, bfs=False):
                     yield self._find_root_tables(select), child
 
+    def find_column_table_mapping(self, sql, file_name):
+        # Parse the SQL query into an expression tree
+        expression = self.parse(sql, file_name)
+
+        # Initialize an empty dictionary to store the column-table mapping
+        column_table_mapping = {}
+
+        # Traverse the expression tree to find all Table and Column expressions
+        for expr in expression:
+            for table_expr in expr.find_all(exp.Table):
+                for column_expr in expr.find_all(exp.Column):
+                    if table_expr.alias == column_expr.table:
```

Comment:
And what if "FROM table AS x JOIN another y USING(foo)"? Make sure there are some good functional tests